src/main.rs

```diff
@@ -72,33 +80,44 @@ async fn main() -> Result<(), APIError> {
         _ => loader.download_model(model_id, None, args.hf_token, args.hf_token_path)?,
     };
 
-    let dtype = match args.dtype.as_deref() {
-        Some("f16") => DType::F16,
-        Some("bf16") => DType::BF16,
-        Some("f32") => DType::F32,
+    let (dtype, dsize) = match args.dtype.as_deref() {
+        Some("f16") => (DType::F16, 2),
+        Some("bf16") => (DType::BF16, 2),
+        Some("f32") => (DType::F32, 4),
         Some(dtype) => panic!("Unsupported dtype {dtype}"),
-        None => DType::BF16,
+        None => (DType::BF16, 2),
     };
 
     let device = candle_examples::device(args.cpu).unwrap();
     let model = loader.load_model(paths, dtype, device)?;
+    let config = model.0.get_model_config();
+    let num_gpu_blocks = args.kvcache_mem * 1024 * 1024
+        / dsize
+        / args.block_size
+        / config.get_num_kv_heads()
+        / config.get_head_size()
+        / config.get_num_hidden_layers()
+        / 2;
+    let cache_config = CacheConfig {
+        block_size: args.block_size,
+        num_gpu_blocks: Some(num_gpu_blocks),
+        num_cpu_blocks: Some(32),
```

Comment:
Can we use the [sysinfo](https://docs.rs/sysinfo/latest/sysinfo/) crate to calculate the total CPU memory (or perhaps allowing the user to set a percentage of CPU memory to use)?

Reply:
> Can we use the [sysinfo](https://docs.rs/sysinfo/latest/sysinfo/) crate to calculate the total CPU memory (or perhaps allowing the user to set a percentage of CPU memory to use)?

Maybe the user what to decide the CPU memory usage for kvcache, or we can use the existing kvcache_mem param for both CPU and GPU?

Reply:
I think the better option is to let the user specify the CPU memory usage, what are your thoughts?

Reply:
> I think the better option is to let the user specify the CPU memory usage, what are your thoughts?

Looks good, perhaps we can add another parameter.